lib/surface/components/form/inputs.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Surface.Components.Form.Inputs do
+  @moduledoc """
+  A wrapper for `Phoenix.HTML.Form.html.inputs_for/3`.
+
+  Additionally, adds the generated form instance that is returned by `inputs_for/3`
+  into the context, making it available to any child input.
+  """
+
+  use Surface.Component
+
+  import Phoenix.HTML.Form
+  import Surface.Components.Form.Utils
+
+  @doc """
+  The parent form.
+
+  It should either be a `Phoenix.HTML.Form` emitted by `form_for` or an atom.
+  """
+  property form, :form
+
+  @doc """
+  The name of the field related to the child inputs.
+  """
+  property for, :atom
+
+  @doc """
+  Extra options for `inputs_for/3`.
+
+  See `Phoenix.HTML.Form.html.inputs_for/4` for the available options.
+  """
+  property opts, :keyword, default: []
+
+  @doc "The code containing the input controls"
+  slot default, props: [:form]
+
+  def render(assigns) do
+    ~H"""
+    <div :for={{ f <- inputs_for(get_form(assigns), @for, @opts) }}>
```

Comment:
I had to use a `<div>` here because currently, there's no way to perform a `:for` without a tag. This is the first case I found where we need this. We need to find a nice way to achieve that. Here's a couple of options:

1. Use `<template :for={{...}}>...</template>` - This is similar to `Vue`'s approach. I believe this will require changes in the compiler/eex_engine to suppress the wrapping `<template>...</template>`.
1. Add a new component `<For>` or `<ForEach>`
1. Add a new directive to suppress the wrapping element. Something like: `<div :for={{}} :unwrap>`

Reply:
The `transform` functionality would allow creating a `For`/`ForEach` component fairly trivially - that's what I had in mind for exploring the `case`/`cond` support

Reply:
`:for` would work on the `Context` component directly:

```
<Context
  :for={{ f <- inputs_for(get_form(assigns), @for, @opts) }}
  set={{ :form, f, scope: Surface.Components.Form }}>
    <slot :props={{ form: f }} />
</Context>
```

Reply:
> The `transform` functionality would allow creating a `For`/`ForEach` component fairly trivially - that's what I had in mind for exploring the `case`/`cond` support

So you think we should go with `#2`. Right?

Reply:
Yeah, I think #2 would be the way to go